CMakeLists.txt

```diff
@@ -1,17 +1,16 @@
-## ---------------------------------------------------------------------
+## ------------------------------------------------------------------------
 ##
+## SPDX-License-Identifier: LGPL-2.1-or-later
```

Comment:
I haven't seen this before, but assume it's a standardized format. Why does it only list the LGPL if we dual-license things?

Reply:
I want to be as precise as possible: just changing the header will not magically dual license the source code. Thus, we simply state the license that effectively applies `LGPL-2.1-or-later` and point to the relicensing effort:
```
## Part of the source code is dual licensed under Apache-2.0 WITH
## LLVM-exception OR LGPL-2.1-or-later. Detailed license information
## governing the source code and code contributions can be found in
## LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
```